packages/cubejs-client-core/src/index.js

```diff
@@ -10,9 +10,17 @@ let mutexCounter = 0;
 
 const MUTEX_ERROR = 'Mutex has been changed';
 
-const mutexPromise = (promise) => new Promise((resolve, reject) => {
-  promise.then(r => resolve(r), e => e !== MUTEX_ERROR && reject(e));
-});
+function mutexPromise(promise) {
```

Comment:
There is a function `asyncDebounce` in `@cubejs-backend/shared`, probably we need to solve this sort of problem in one way.